flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptorFactory.java

```diff
@@ -216,14 +221,16 @@ private ShuffleDescriptorAndIndex[] computeConsumedPartitionShuffleDescriptors(
     }
 
     private MaybeOffloaded<ShuffleDescriptorAndIndex[]> serializeAndTryOffloadShuffleDescriptor(
-            ShuffleDescriptorAndIndex[] shuffleDescriptors) throws IOException {
+            ShuffleDescriptorAndIndex[] shuffleDescriptors, boolean shouldOffload)
+            throws IOException {
 
         final CompressedSerializedValue<ShuffleDescriptorAndIndex[]> compressedSerializedValue =
                 CompressedSerializedValue.fromObject(shuffleDescriptors);
 
         final Either<SerializedValue<ShuffleDescriptorAndIndex[]>, PermanentBlobKey>
                 serializedValueOrBlobKey =
-                        BlobWriter.tryOffload(compressedSerializedValue, jobID, blobWriter);
+                        BlobWriter.tryOffload(
```

Comment:
Since `BlobWriter` don't know the logic of task deployment, here we could say:

```java
final Either<SerializedValue<ShuffleDescriptorAndIndex[]>, PermanentBlobKey>
        serializedValueOrBlobKey =
                shouldOffload(numConsumers)
                        ? BlobWriter.offloadWithException(
                                compressedSerializedValue, jobID, blobWriter)
                        : Either.Left(compressedSerializedValue);
```